core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -319,15 +320,7 @@ private void commitCreateTransaction() {
 
     } catch (RuntimeException e) {
       // the commit failed and no files were committed. clean up each update.
-      Tasks.foreach(updates)
-          .suppressFailureWhenFinished()
-          .run(
-              update -> {
-                if (update instanceof SnapshotProducer) {
-                  ((SnapshotProducer) update).cleanAll();
-                }
-              });
-
+      computeUncommittedFiles();
```

Comment:
I think this needs to be refactored more. If the `RuntimeException` was something like a 500 error for a create table or replace table transaction, then the table may have been created or replaced. For strict cleanup, I think we still need to check for a cleanable exception in those cases.

I think these should be:
```java
} catch (RuntimeException e) {
  if (!ops.requiresStrictCleanup() || e instanceof CleanableFailure) {
    cleanUpOnCommitFailure();
  }
}

Tasks.foreach(deletedFiles).run(ops.io()::delete);
```

Reply:
Right, it's possible that during the create/replace we get a 500 but the create/replace actually was successful. So we still need the cleanable failure check. I realized I'm also missing tests for create/replace/simple transactions so I'll add those

Reply:
I've updated. If I'm not missing anything we would still want to preserve the behavior of cleaning up the files in the `finally` block because that set of deleted files would the files from previous commit attempts (this was already happening regardless of commit outcome, including CommitUnknownException). We should be able to clean those files up in any situation.